### PR TITLE
Stripe UI - create card payment method

### PR DESCRIPTION
**Pull request by ClaudeROSSI**
A draft of the ui has been added to the "template" module
- It allows user to link their card to their account ( as a default payment method)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,7 +31,6 @@ speed-measure-plugin*.json
 
 # IDE - VSCode
 .vscode/*
-!.vscode/settings.json
```
Comment (LucasBrazi06) [inline .gitignore:34]: Have you removed it on purpose?

File: src/app/pages/template/stripe-template/stripe-template.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.spec.ts:1]: No need to generate this file I think.

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  styleUrls: ['./stripe-template.component.scss']
+})
+
+export class StripeTemplateComponent implements OnInit {
+
+  // Billing & Billing Information
+  public isActive = false;
+  public billingSettings!: BillingSettings;
+
+  @ViewChild('cardInfo', { static: true }) cardInfo: ElementRef;
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:32]: Needed?

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  ) {
+    this.isActive = this.componentService.isActive(TenantComponents.BILLING);
+  }
+
+  ngOnInit(): void {
+    this.initialize();
+  }
+
+  async initialize(): Promise<void> {
+
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:57]: Blank lines :-P

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      this.spinnerService.show();
+      this.billingSettings = await this.loadBillingConfiguration();
+      if ( !this.billingSettings || !this.billingSettings.stripe ) {
+        this.feedback = "Stripe configuration is not set for the current tenant";
+      } else {
+        this.stripe = await this.initializeStripe(this.billingSettings.stripe.publicKey);
+        this.initializeCardElements();
+      }
+    }
+    catch (error) {
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:69]: 'catch' after the curly brace:
} catch (error) {

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+      Utils.handleHttpError(error, this.router, this.messageService, this.centralServerService, 'general.unexpected_error_backend');
+    }
+    finally {
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:72]: Same as above

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+      this.spinnerService.hide();
+    }
+
+  }
+
+  async initializeStripe(publicKey: string) : Promise<Stripe> {
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:78]: Usually we always specify the visibility even if it's public.

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+    loadStripe.setLoadParameters({ advancedFraudSignals: false })
+    return await loadStripe(publicKey);
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:81]: No need to specify 'await', you can return the promise right away.

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    }
+  }
+
+  linkCardToAccount() {
+    this.doCreatePaymentMethod();
+  }
+
+  async doCreatePaymentMethod() {
+
+    const operationResult: any = await this.createPaymentMethod()
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:116]: Semi column at the end of the line

File: src/app/pages/template/stripe-template/stripe-template.component.ts [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    if (operationResult.error) {
+      // Operation failed
+      this.feedback = operationResult.error.message;
+    } else {
+      // Operation succeeded
+      this.feedback = operationResult;
+
```
Comment (LucasBrazi06) [inline src/app/pages/template/stripe-template/stripe-template.component.ts:123]: Can be also done this way:
this.feedback = operationResult.error ? operationResult.error.message : operationResult;

**Review by LucasBrazi06 — COMMENTED** [review]
Few comments I think due to tests :-)

File: .gitignore (hunk 1 above)
Comment (ClaudeROSSI) [reply .gitignore:34]: I did it on purpose ... I have personal colour settings per project. 
In other words, I need to change the colour settings without impacting the other developers.
are you ok?

File: src/app/pages/template/stripe-template/stripe-template.component.spec.ts (hunk 2 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.spec.ts:1]: right!

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 4 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:58]: done ... removed all blank lines

File: .vscode/launch.json [hunk 11]
```diff
@@ -7,9 +7,17 @@
     {
       "type": "chrome",
       "request": "launch",
-      "name": "Launch Chrome against localhost",
+      "name": "tenant: slf",
```
[the remark this answers is not in the document]
Comment (ClaudeROSSI) [reply .vscode/launch.json:10]: I would like to git ignore that launch.json config ... to make sure each developer can have his own settings.
Is it ok with you?

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 5 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:70]: ok

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 6 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:73]: ok

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 7 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:79]: done ... almost all private

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 8 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts]: Temporary code ... I need to review the initialisation. Lazy loading of stripe will require a dedicated service.
Still under construction!

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 10 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:137]: temporary code ... only used in the sandbox view

File: .gitignore (hunk 1 above)
Comment (LucasBrazi06) [reply .gitignore:34]: @jerome-benoit Ok for you?

File: .vscode/launch.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply .vscode/launch.json:10]: @jerome-benoit Ok for you?

File: src/app/pages/template/stripe-template/stripe-template.component.ts (hunk 3 above)
Comment (ClaudeROSSI) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:22]: which one? most have been moved to the stripe service.
cardInfo is somehow  the form from stripe

Comment (LucasBrazi06) [reply src/app/pages/template/stripe-template/stripe-template.component.ts:22]: The whole line
`@ViewChild('cardInfo', { static: true }) cardInfo: ElementRef;`

File: .gitignore (hunk 1 above)
Comment (jerome-benoit) [reply .gitignore:34]: Yes, we've done that setting with Claude.

File: .vscode/launch.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply .vscode/launch.json:10]: Yes, we've done that setting Claude. 

That debate remains on if it should be a vscode template settings file or like it is right now. 

@ClaudeROSSI I remember there's was more tunables added to make debugging works in vscode? Are you sure you've pushed the right working version in that PR? 

And I think that fix should be go in another PR and merged ASAP, it make vscode debug works again on the dashboard

[the remark this answers is not in the document]
Comment (ClaudeROSSI) [reply .vscode/launch.json:10]: Now that I have a macbook it woks perfectly.
No need for a different config. However  each developer may need some specific settings (on windows for instance).
So I would propose to git ignore the launch.json and just push a launch-template.json for new comers.

Same story backend side. I had to change the settings for debugging mocha tests properly (with a different port to make it possible to debug both the server and the tests at the same time). so a template makes sense here as well.

[the remark this answers is not in the document]
Comment (jerome-benoit) [reply .vscode/launch.json:10]: > Now that I have a macbook it everything is fine.
> No need for a different config. However each developer may need some specific settings (on windows for instance).
> So I would propose to git ignore the launch.json and just push a launch-template.json for new comers.
> 
> Same story backend side. I had to change the settings for debugging mocha tests properly (with a different port to make it possible to debug both the server and the tests at the same time). so a template makes sense here as well.

Fair enough. Could you please make the PRs?

**Comment by ClaudeROSSI** [comment]
changes will be provided via a different branch from Caroline

**Pull request closed without merging**